cachi2/core/package_managers/rpm/main.py

```diff
@@ -214,3 +215,76 @@ def _generate_sbom_components(files_metadata: dict[Path, Any]) -> list[Component
+
+
+def inject_files_post(*args: Any, **kwargs: Any) -> None:
+    """Run extra tasks for the rpm package manager (callback method) within `inject-files` cmd."""
+    if "from_output_dir" in kwargs and "for_output_dir" in kwargs:
+        from_output_dir = kwargs["from_output_dir"]
+        for_output_dir = kwargs["for_output_dir"]
+
+        if Path.exists(from_output_dir.joinpath(DEFAULT_PACKAGE_DIR)):
+            generate_repos(from_output_dir)
+            generate_repofiles(from_output_dir, for_output_dir)
+
+
+def generate_repos(from_output_dir: Path) -> None:
+    """Search structure for all repoid dirs and create repository metadata \
+    out of its RPMs (and SRPMs)."""
+    package_dir = from_output_dir.joinpath(DEFAULT_PACKAGE_DIR)
+    for arch in package_dir.iterdir():
+        if not arch.is_dir():
+            continue
+        for repoid in arch.iterdir():
+            if not repoid.is_dir() or repoid.name == "repos.d":
+                continue
+            createrepo(repoid.name, repoid)
+
+
+def createrepo(reponame: str, repodir: Path) -> None:
+    """Execute the createrepo utility."""
+    log.info(f"Creating repository metadata for repoid '{reponame}': {repodir}")
+    cmd = ["createrepo_c", str(repodir)]
+    log.debug("$ " + shlex.join(cmd))
+    stdout = run_cmd(cmd, params={})
+    log.debug(stdout)
+
+
+def generate_repofiles(from_output_dir: Path, for_output_dir: Path) -> None:
+    """
+    Generate templates of repofiles for all arches.
+
+    Search structure at 'path' and generate one repofile content for each arch.
+    Each repofile contains all arch's repoids (including repoids with source RPMs).
+    Repofile (cachi2.repo) for particular arch will be stored in arch's dir in 'repos.d' subdir.
+    Repofiles are not directly created from the templates in this method - templates are stored
+    in the project file.
+    """
+    package_dir = from_output_dir.joinpath(DEFAULT_PACKAGE_DIR)
+    for arch in package_dir.iterdir():
+        if not arch.is_dir():
+            continue
+        log.debug(f"Preparing repofile content for arch '{arch.name}'")
+        content = ""
+        for repoid in arch.iterdir():
+            if not repoid.is_dir() or repoid.name == "repos.d":
+                continue
+            localpath = for_output_dir.joinpath(DEFAULT_PACKAGE_DIR, arch.name, repoid.name)
+            content += f"[{repoid.name}]\n"
+            content += f"baseurl=file://{localpath}\n"
+            content += "gpgcheck=1\n"
+            if repoid.name.startswith("cachi2-"):  # repoid directory matches the internal repoid
+                content += (
+                    "name=Generated repository containing all packages unaffiliated "
+                    "with any official repository\n"
+                )
+        if content:
+            repo_file_path = arch.joinpath("repos.d", "cachi2.repo")
+            if repo_file_path.exists():
+                log.info(f"Overwriting {repo_file_path}")
```

Comment:
The idea comes from:
cachi2/interface/cli.py:334
where it is an info message.
I thought the deleting was related to `fetch-deps`. This is related to `inject-files` and here, I think, we could have this situation.

Reply:
> I thought the deleting was related to fetch-deps. This is related to inject-files and here, I think, we could have this situation.

True, my bad then.

>  cachi2/interface/cli.py:334

Well, IMO incorrect on its own, but since there's a precedent, you can leave this as is.